### PR TITLE
1st contribution

### PR DESCRIPTION
**Pull request by uklotzde**
Some basic project settings with one open TODO in lib.rs.

The only notable change is the renaming in the process_bit() functions. I found the existing naming very confusing that referred to the expected position values instead of the time when they were actually obtain, i.e. before/after updating the bitstream.

### CODE REVIEW

**Comment by uklotzde** [comment]
Informal naming conventions: https://rust-lang.github.io/api-guidelines/

**Comment by uklotzde** [comment]
Force push: The new debug assertions in process_bit() were wrong. Everything is possible, independent of the current direction.

File: src/timecode.rs [hunk 1]
```diff
@@ -137,13 +135,17 @@ impl Timecode {
         //  ╰(1)╯   ╰───╯   ╰───╯  forwards, otherwise it's playing backwards.
         //
         if primary_crossed_zero {
-            self.direction = if self.primary_channel.wave_cycle_status == self.secondary_channel.wave_cycle_status {
+            self.direction = if self.primary_channel.wave_cycle_status
+                == self.secondary_channel.wave_cycle_status
+            {
```
Comment (Holzhaus) [inline src/timecode.rs:146]: Is this rustfmt's default setting? I used "opening bracket on same line" everywhere else.

**Review by Holzhaus — COMMENTED** [review]
Thanks. I'm considering relicensing to MPL 2.0 (as [discussed here](https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/Vinyl.20Control.20library/near/226753725)), do I have your permission to do that?

File: src/timecode.rs (hunk 1 above)
Comment (uklotzde) [reply src/timecode.rs:146]: Yes, `cargo fmt` in project directory. Otherwise include a custom .rustfmt file.

**Comment by uklotzde** [comment]
> Thanks. I'm considering relicensing to MPL 2.0 (as [discussed here](https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/Vinyl.20Control.20library/near/226753725)), do I have your permission to do that?

Sure. At your choice, whatever you finally choose ;)

**Comment by uklotzde** [comment]
We can use the checks that I prepared here: https://github.com/aoide-org/aoide-rs/blob/development/.github/workflows/continuous_integration.yaml

Then the build will fail if the formatting does not follow rustfmt (as is currently the case with the last commit).

File: src/timecode.rs (hunk 1 above)
Comment (Holzhaus) [reply src/timecode.rs:146]: I was wrong, I use the same style as you use here in the code below. :shrug:

**Comment by Holzhaus** [comment]
> We can use the checks that I prepared here: https://github.com/aoide-org/aoide-rs/blob/development/.github/workflows/continuous_integration.yaml
> 
> Then the build will fail if the formatting does not follow rustfmt (as is currently the case with the last commit).

I'm going to copy my pre-commit hooks from here, then it will run rustfmt and clippy when committing: https://github.com/Holzhaus/triseratops/blob/main/.pre-commit-config.yaml

**Comment by uklotzde** [comment]
Thanks for the reminder ;) I also need to get the pre-commit checks in place.

**Pull request merged**